metrics.go

```diff
@@ -64,8 +85,18 @@ func NewMetrics(conf *metrics.Config, sink metrics.MetricSink) (err error) {
 }
 
 // SetMetrics initializes goa's metrics instance with the supplied metrics adapter interface.
-func SetMetrics(m *metrics.Metrics) {
-	metriks.Store(m)
+func SetMetrics(m Collector) {
+	metriksMu.Lock()
+	metriks = m
+	metriksMu.Unlock()
+}
+
+// GetMetrics returns goa's metrics collector adapter interface.
+func GetMetrics() Collector {
+	metriksMu.Lock()
```

Comment:
Could we use a RWMutex instead and simply use a read lock here? otherwise this causes all concurrent requests to have to be synchronized.